mcx/jax.py

```diff
@@ -76,49 +76,45 @@ def unravel_list(arr):
     return flat, unravel_list
 
 
-def define_tqdm(num_samples):
-    global tqdm_pbar
-    tqdm_pbar = tqdm(range(num_samples))
-
-
-def close_tqdm():
-    tqdm_pbar.close()
-
-
-def set_description_tqdm(message):
-    tqdm_pbar.set_description(
-        message,
-        refresh=False,
-    )
-
-
-def _update_tqdm(arg, transform):
-    tqdm_pbar.update(arg)
-
-
-@jit
-def _progress_bar(arg, result):
-    """Updates tqdm progress bar of a scan/loop only if the iteration number is a multiple of the print_rate
-    Usage: carry = progress_bar((iter_num, print_rate), carry)
+def progress_bar_factory(num_samples):
```

Comment:
@rlouf is this what you meant?

Reply:
It looks great! Maybe we could simplify this a little bit before merging:

1. Define the `tqdm_pbar` in `sample _scan`;
2. Pass this `tqdm_bar` as an argument to the factory;
3. Then the instructions of `set_tqdm_description` and `close_tqdm` are moved to `sample_scan`; they don't need to be included in a function anymore. 

Below the changes I would make.